From: Ansgar Burchardt <ansgar@43-1.org>
Subject: Disable tests that require a running database
Date: Mon, 27 Jul 2009 23:42:21 +0200

Disable tests that require a MySQL server.  Also add a test that makes sure the
module can at least be loaded.
--- libclass-dbi-mysql-perl.orig/t/mysql.t
+++ libclass-dbi-mysql-perl/t/mysql.t
@@ -3,6 +3,8 @@
 use strict;
 use Test::More;
 
+plan skip_all => "Database tests disabled." unless $ENV{TEST_DATABASE};
+
 my $HAVE_TP = eval { require Time::Piece::MySQL };
 
 plan tests => 8;
@@ -37,7 +39,7 @@
 __PACKAGE__->set_up_table;
 __PACKAGE__->autoinflate(dates => 'Time::Piece') if $HAVE_TP;
 
-END { __PACKAGE__->drop_table }
+END { __PACKAGE__->drop_table if $ENV{TEST_DATABASE}; }
 
 #-------------------------------------------------------------------------
 
--- /dev/null
+++ libclass-dbi-mysql-perl/t/01use.t
@@ -0,0 +1,3 @@
+#! /usr/bin/perl
+use Test::More tests => 1;
+use_ok "Class::DBI::mysql";
